### PR TITLE
Clean routes on node delete

### PR DESCRIPTION
**Pull request by Thiblef**

### CODE REVIEW

File: BackofficeBundle/EventSubscriber/UpdateNodeRedirectionSubscriber.php [hunk 1]
```diff
@@ -65,12 +65,25 @@ public function updateRedirection(NodeEvent $event)
...
+    }
+
+    /**
      * @return array The event names to listen to
      */
     public static function getSubscribedEvents()
     {
         return array(
             NodeEvents::NODE_CHANGE_STATUS => 'updateRedirection',
+            NodeEvents::NODE_RESTORE => 'updateRedirectionRoutes',
```
Comment (SOLenG) [inline BackofficeBundle/EventSubscriber/UpdateNodeRedirectionSubscriber.php]: the term "restore" for a function called on update disturbs me

Comment (nicolasThal) [inline BackofficeBundle/EventSubscriber/UpdateNodeRedirectionSubscriber.php]: @SOLenG the `restore` term is linked to the name of the event launched. The `update` term is linked to the performed action.

IMHO, I think that this association is ok

File: BackofficeBundle/EventSubscriber/UpdateNodeRedirectionSubscriber.php [hunk 2]
```diff
@@ -65,12 +65,25 @@ public function updateRedirection(NodeEvent $event)
     }
 
     /**
+     * @param NodeEvent $event
+     */
+    public function updateRedirectionRoutes(NodeEvent $event)
```
Comment (itkg-alavieille) [inline BackofficeBundle/EventSubscriber/UpdateNodeRedirectionSubscriber.php]: can you add test for this method in UpdateNodeRedirectionSubscriberTest

File: BackofficeBundle/EventSubscriber/UpdateNodeRedirectionSubscriber.php (hunk 1 above)
Comment (SOLenG) [inline BackofficeBundle/EventSubscriber/UpdateNodeRedirectionSubscriber.php]: My mistake, I had not done enough,besides that respects the structure of Subscribers

**Comment by Thiblef** [comment]
@nicolasThal @SOLenG @itkg-alavieille 
Added tests

**Comment by itkg-alavieille** [comment]
:+1:

**Comment by SOLenG** [comment]
:+1:

**Pull request merged**